src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -612,6 +627,13 @@ private static boolean unnecessaryParenAroundOperators(DetailAST ast) {
         final boolean isConditional = TokenUtil.isOfType(type, CONDITIONALS_AND_RELATIONAL);
         boolean result = TokenUtil.isOfType(parentType, CONDITIONALS_AND_RELATIONAL);
         if (isConditional) {
+            if (parentType == TokenTypes.BAND || parentType == TokenTypes.BXOR
+                    || parentType == TokenTypes.BOR) {
```

Comment:
Please create an array as we have done for `CONDITIONALS_AND_RELATIONAL`, let's call it `BITWISE_BINARY_OPERATORS`.  Then we can use `TokenUtil.isOfType` here.

Reply:
Code here gets weird as we consider operator precedence; please add a comment here about why parenthesis are unnecessary in this case.

Reply:
done